app/models/miq_queue.rb

```diff
@@ -173,6 +173,9 @@ def self.put(options)
   def self.submit_job(options)
     service = options.delete(:service) || "generic"
     resource = options.delete(:affinity)
+
+    return unless resource_enabled?(resource)
```

Comment:
I'm confused, what about places that do `MiqQueue.put` or other callers of `.put`?  Why is `submit_job` singled out here?  I'd think we'd need it everywhere we put something on the queue.